DiyorMarket.MVC/Lesson11/Views/Products/Index.cshtml

```diff
@@ -1,136 +1,130 @@
-        </div>
-
-
-    <div class="d-flex justify-content-between">
-        <div class="p-2" style="font-size: 14px;">
-
-            @if (ViewBag.HasPreviousPage)
-            {
-                <a class="btn btn-outline-primary rounded-left" href="@Url.Action("Index", new { pageNumber = 1 })">&lt;&lt;</a>
-            }
-
-            @if (ViewBag.HasPreviousPage)
-            {
-                <a class="btn btn-outline-primary rounded-right" href="@Url.Action("Index", new { pageNumber = ViewBag.CurrentPage - 1})">@(@ViewBag.CurrentPage - 1)</a>
-            }
-
-            <button type="button" class="btn btn-primary rounded-right" disabled>@ViewBag.CurrentPage</button>
-
-            @if (ViewBag.HasNextPage)
-            {
-                <a class="btn btn-outline-primary rounded-right" href="@Url.Action("Index", new { pageNumber = ViewBag.CurrentPage + 1})">@(@ViewBag.CurrentPage + 1)</a>
-            }
-
-            @if (ViewBag.CurrentPage <= 1)
-            {
-                <a class="btn btn-outline-primary rounded-right" href="@Url.Action("Index", new { pageNumber = ViewBag.CurrentPage + 2})">@(@ViewBag.CurrentPage + 2)</a>
-            }
-
-            @if (ViewBag.HasNextPage)
-            {
-                <a class="btn btn-outline-primary rounded-right" href="@Url.Action("Index", new { pageNumber = ViewBag.PageCount })">&gt;&gt;</a>
-            }
+    </div>
+</form>
+
+
+<div class="row">
+</div>
+<div class="col-12">
+    <ejs-grid id="products-list"
+              dataSource="@ViewBag.Products"
+              gridLines="Vertical"
+              allowSorting="true">
+        <e-grid-columns>
+            <e-grid-column headerText="Id" field="Id" template="#template" typeof="Text"></e-grid-column>
+            <e-grid-column headerText="Name" field="Name" typeof="text"></e-grid-column>
+            <e-grid-column headerText="Description" field="Description" typeof="text"></e-grid-column>
+            <e-grid-column headerText="Sale Price" field="SalePrice" typeof="number" format="C2"></e-grid-column>
+            <e-grid-column headerText="Supply Price" field="SupplyPrice" typeof="number" format="C2"></e-grid-column>
+            <e-grid-column headerText="Expire Date" field="ExpireDate" format="dd/MM/yyyy, HH:mm:ss" typeof="text"></e-grid-column>
+            <e-grid-column headerText="Category" field="Category.Name" typeof="text"></e-grid-column>
+        </e-grid-columns>
+    </ejs-grid>
+</div>
+
+
+<div class="d-flex justify-content-between">
+    <div class="p-2" style="font-size: 14px;">
+
+        @if (ViewBag.HasPreviousPage)
+        {
+            <a class="btn btn-outline-primary rounded-left" href="@Url.Action("Index", new { pageNumber = 1 })">&lt;&lt;</a>
+        }
+
+        @if (ViewBag.HasPreviousPage)
+        {
+            <a class="btn btn-outline-primary rounded-right" href="@Url.Action("Index", new { pageNumber = ViewBag.CurrentPage - 1})">@(@ViewBag.CurrentPage - 1)</a>
```

Comment:
We can send **PreviousPage** from backend to ViewBag directly in order to get rid of **ViewBag.Currentage - 1** logic. Someone can forget it or change to **ViewBag.CurrentPage - 2** and so on.